IPython/core/display.py

```diff
@@ -467,16 +467,26 @@ def _repr_markdown_(self):
 class Math(TextDisplayObject):
 
     def _repr_latex_(self):
-        s = self.data.strip('$')
-        return "$$%s$$" % s
+        if self.format == u'latex' or not self.format:
+            s = self.data.strip('$')
+            return "$$%s$$" % s
+        
+    def _repr_html_(self):
+        if self.format == u'mathml':
+            return self.data
```

Comment:
When given math for latex, we don't expect it to include the `$$` wrapper, that is `Math('...')` replaces the `$$` as the wrapper for the math inside. To match that for mathml might be to *add* the `<math>` tag, which might make it a bit harder to identify math by regex.

Right now, you can do:

```python
Math('a=5')
```

so I would expect

```python
Math('a=5', format='mathml')
```

to probably work as well.